src/core/thread/mle_tlvs.hpp

```diff
@@ -1466,7 +1466,7 @@ class AddressRegistrationTlv: public Tlv
     }
 
 private:
-    AddressRegistrationEntry mAddresses[4];
+    AddressRegistrationEntry mAddresses[OPENTHREAD_CONFIG_IP_ADDRS_PER_CHILD];
```

Comment:
Excellent catch! I think i run on this problem before, but didn't have time to debug it.